File: src/main/java/de/tectoast/emolga/utils/records/ModalConfiguration.java
package de.tectoast.emolga.utils.records;

import de.tectoast.emolga.utils.automation.structure.ModalConfigurator;

import java.util.function.Supplier;

@SuppressWarnings("unused")
public record ModalConfiguration(String name, Supplier<ModalConfigurator> configuratorSupplier) {
    public ModalConfigurator configurator() {
        return configuratorSupplier.get();
    }
}

File: src/main/kotlin/de/tectoast/emolga/utils/records/ModalConfiguration.kt
package de.tectoast.emolga.utils.records

import de.tectoast.emolga.utils.automation.structure.ModalConfigurator
import java.util.function.Supplier

class ModalConfiguration(val name: String, private val configuratorSupplier: Supplier<ModalConfigurator>) {
    fun configurator(): ModalConfigurator {
        return configuratorSupplier.get()
    }
}